src/drivers/distance_sensor/leddar_one/leddar_one.cpp

```diff
@@ -0,0 +1,564 @@
+			PX4_ERR("No sample available in %s", DEVICE_PATH);
+			return PX4_ERROR;
+		}
+
+		bool valid = false;
+
+		if (report.current_distance > report.min_distance
+		    && report.current_distance < report.max_distance) {
+			valid = true;
+		}
+
+		printf("valid: %u\n", valid);
+		printf("distance: %0.3fm\n", (double)report.current_distance);
+		printf("time: %llu\n", report.timestamp);
+
+	} else {
+		help();
+		return PX4_ERROR;
+	}
+
+	return PX4_OK;
+}
+
+leddar_one::leddar_one(const char *device_path, const char *serial_port, uint8_t rotation):
+	CDev(NAME, device_path),
+	_rotation(rotation),
+	_collect_timeout_perf(perf_alloc(PC_COUNT, "leddar_one_collect_timeout")),
+	_comm_error(perf_alloc(PC_COUNT, "leddar_one_comm_errors")),
+	_sample_perf(perf_alloc(PC_ELAPSED, "leddar_one_sample"))
+{
+	memset(&_work, 0, sizeof(_work));
+	_serial_port = strdup(serial_port);
+}
+
+leddar_one::~leddar_one()
+{
+	free((char *)_serial_port);
+
+	if (_fd > -1) {
+		::close(_fd);
+	}
+
+	if (_reports) {
+		delete _reports;
+	}
+
+	if (_topic) {
+		orb_unadvertise(_topic);
+	}
+
+	perf_free(_sample_perf);
+	perf_free(_collect_timeout_perf);
+	perf_free(_comm_error);
+}
+
+int leddar_one::_fd_open()
+{
+	if (!_serial_port) {
+		return -1;
+	}
+
+	_fd = ::open(_serial_port, O_RDWR | O_NOCTTY | O_NONBLOCK);
+
+	if (_fd < 0) {
+		return -1;
+	}
+
+	struct termios config;
+
+	int r = tcgetattr(_fd, &config);
+
+	if (r) {
+		PX4_ERR("Unable to get termios from %s.", _serial_port);
+		goto error;
+	}
+
+	/* clear: data bit size, two stop bits, parity, flow control */
+	config.c_cflag &= ~(CSIZE | CSTOPB | PARENB | CCTS_OFLOW | CRTS_IFLOW);
+	/* set: 8 data bits, enable receiver, ignore modem status lines */
+	config.c_cflag |= (CS8 | CREAD | CLOCAL);
+	/* turn off output processing */
+	config.c_oflag = 0;
+	/* clear: echo, echo new line, canonical input and extended input */
+	config.c_lflag &= (ECHO | ECHONL | ICANON | IEXTEN);
+
+	r = cfsetispeed(&config, B115200);
+	r |= cfsetospeed(&config, B115200);
+
+	if (r) {
+		PX4_ERR("Unable to set baudrate");
+		goto error;
+	}
+
+	r = tcsetattr(_fd, TCSANOW, &config);
+
+	if (r) {
+		PX4_ERR("Unable to set termios to %s", _serial_port);
+		goto error;
+	}
+
+	return PX4_OK;
+
+error:
+	::close(_fd);
+	_fd = -1;
+	return PX4_ERROR;
+}
+
+int leddar_one::init()
+{
+	if (CDev::init()) {
+		PX4_ERR("Unable to initialize device\n");
+		return -1;
+	}
+
+	_reports = new ringbuffer::RingBuffer(2, sizeof(distance_sensor_s));
+
+	if (!_reports) {
+		PX4_ERR("No memory to allocate RingBuffer");
+		return -1;
+	}
+
+	if (_fd_open()) {
+		return PX4_ERROR;
+	}
+
+	hrt_abstime timeout_usec, now;
+
+	for (now = hrt_absolute_time(), timeout_usec = now + PROBE_USEC_TIMEOUT;
+	     now < timeout_usec;
+	     now = hrt_absolute_time()) {
+
+		if (_cycle() > 0) {
+			printf(NAME " found\n");
+			return PX4_OK;
+		}
+
+		usleep(1000);
+	}
+
+	PX4_ERR("No readings from " NAME);
+
+	return PX4_ERROR;
+}
+
+void
+leddar_one::stop()
```

Comment:
I don't think this is being executed, right?

Reply:
Fixed